src/plugins_3rdparty/primer3/src/Primer3Dialog.cpp

```diff
@@ -595,172 +747,280 @@ void Primer3Dialog::sl_saveSettings() {
-    diagSettings.setValue("PRIMER_LEFT_INPUT", ui.edit_PRIMER_LEFT_INPUT->text());
-    diagSettings.setValue("PRIMER_RIGHT_INPUT", ui.edit_PRIMER_RIGHT_INPUT->text());
-    diagSettings.setValue("PRIMER_INTERNAL_OLIGO_INPUT", ui.edit_PRIMER_INTERNAL_OLIGO_INPUT->text());
+    U2OpStatusImpl os;
+    stream << "SEQUENCE_TEMPLATE=" << context->getSequenceObject()->getWholeSequenceData(os) << endl;
+    stream << "SEQUENCE_ID=" << context->getSequenceObject()->getSequenceName() << endl;
 
-    diagSettings.setValue("PRIMER_TM_SANTALUCIA", ui.combobox_PRIMER_TM_SANTALUCIA->currentIndex());
-    diagSettings.setValue("PRIMER_SALT_CORRECTIONS", ui.combobox_PRIMER_SALT_CORRECTIONS->currentIndex());
-
-    diagSettings.setValue("PRIMER_LIBERAL_BASE", ui.checkbox_PRIMER_LIBERAL_BASE->isChecked());
-    diagSettings.setValue("PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS", ui.checkbox_PRIMER_LIB_AMBIGUITY_CODES_CONSENSUS->isChecked());
-    diagSettings.setValue("PRIMER_LOWERCASE_MASKING", ui.checkbox_PRIMER_LOWERCASE_MASKING->isChecked());
-    diagSettings.setValue("PICK_HYBRO", ui.checkbox_PICK_HYBRO->isChecked());
-    diagSettings.setValue("PICK_LEFT", ui.checkbox_PICK_LEFT->isChecked());
-    diagSettings.setValue("PICK_RIGHT", ui.checkbox_PICK_RIGHT->isChecked());
+    auto qualityText = edit_SEQUENCE_QUALITY->toPlainText();
+    if (!qualityText.isEmpty()) {
+        stream << "SEQUENCE_QUALITY=" << qualityText << endl;
+    }
+    
+    stream << "PRIMER_TASK=" << edit_PRIMER_TASK->currentText() << endl;
+
+    stream << "PRIMER_TM_FORMULA=" << combobox_PRIMER_TM_FORMULA->currentIndex() << endl;
+    stream << "PRIMER_SALT_CORRECTIONS=" << combobox_PRIMER_SALT_CORRECTIONS->currentIndex() << endl;
+    
+    QString pathPrimerMisprimingLibrary;
+    QString pathPrimerInternalOligoLibrary;
+    for (const auto& lib : repeatLibraries) {
+        if (lib.first == combobox_PRIMER_MISPRIMING_LIBRARY->currentText() && !lib.second.isEmpty()) {
+            QFileInfo fi(lib.second);
+            stream << "PRIMER_MISPRIMING_LIBRARY=" << fi.fileName() << endl;
+        }
+        if (lib.first == combobox_PRIMER_INTERNAL_MISHYB_LIBRARY->currentText() && !lib.second.isEmpty()) {
+            QFileInfo fi(lib.second);
+            stream << "PRIMER_INTERNAL_MISHYB_LIBRARY=" << fi.fileName() << endl;
+        }
+    }
 
-    diagSettings.setValue("PRIMER_MISPRIMING_LIBRARY", ui.combobox_PRIMER_MISPRIMING_LIBRARY->currentIndex());
-    diagSettings.setValue("PRIMER_INTERNAL_OLIGO_MISHYB_LIBRARY", ui.combobox_PRIMER_INTERNAL_OLIGO_MISHYB_LIBRARY->currentIndex());
+    stream << "=" << endl;
 
-    diagSettings.endGroup();
+    file.close();
 }
 
 void Primer3Dialog::sl_loadSettings() {
     LastUsedDirHelper lod;
-    lod.url = U2FileDialog::getOpenFileName(this, tr("Load settings"), lod.dir, "Text files (*.txt)");
+    lod.url = U2FileDialog::getOpenFileName(this, tr("Load settings"), lod.dir, tr("All files") + " ( * );;" + tr("Text files") + "(*.txt)");
```

Comment:
Use `QString FileFilters::withAllFilesFilter(const QStringList& filters) {` to have `All files` added in the standard way.

You can also use text format file filter for txt files.

Reply:
Ok in this case we have `static QString createSingleFileFilter(const QString& name, const QStringList& extensions, bool addGzipVariant);`